components/tidb_query/src/codec/datum.rs

```diff
@@ -164,11 +165,16 @@ impl Datum {
             Datum::I64(i) => cmp_f64(i as f64, f),
             Datum::U64(u) => cmp_f64(u as f64, f),
             Datum::F64(ff) => cmp_f64(ff, f),
-            Datum::Bytes(ref bs) => cmp_f64(bs.convert(ctx)?, f),
+            Datum::Bytes(ref bs) => cmp_f64(bs.as_slice().convert(ctx)?, f),
             Datum::Dec(ref d) => cmp_f64(d.convert(ctx)?, f),
             Datum::Dur(ref d) => cmp_f64(d.to_secs_f64(), f),
-            Datum::Time(ref t) => cmp_f64(t.convert(ctx)?, f),
-            Datum::Json(ref json) => Datum::F64(f).cmp_json(ctx, json),
+            Datum::Time(ref t) => {
+                // TODO, here is not exactly same as TiDB's
+                let r: Decimal = t.convert(ctx)?;
```

Comment:
Why not convert from time to f64 directly?

Reply:
Because TiDB do it like this.

Reply:
We don't need to convert time to decimal first.